src/contexts/applicationContext.tsx

```diff
@@ -34,27 +35,41 @@ function ApplicationProvider({ children }: ApplicationProviderProps) {
   const [viewState, setViewState] = useState<ViewState>(ViewState.Overview)
   const [initComplete, setInitComplete] = useState(false)
   const [collections, setCollections] = useState<Collection[]>([])
+  const [searchValue, setSearchValue] = useState('')
+  const [isSearching, setIsSearching] = useState<boolean>(false)
```

Comment:
The state value for searchValue exists because I use the value in multiple files at ones to change the display of certain elements.

I'll get rid of the isSearching and can move the searchValue a bit closer, but not sure if theres a better idea on how to sync up the search term when read by multiple functions at once

Reply:
The searchValue should be passed as props to the components that its relevant to.

The isSearching itself isn't a problem, it's more the fact of using a state variable for it, you could just do `const isSearching = searchValue.trim().length !== 0` and then you don't have to worry about making sure you update it everytime. 

Then if you're worried about the execution time for that, you just wrap it in a useMemo

Reply:
Reworked: [69f2c76](https://github.com/WootingKb/wooting-macros/pull/194/commits/69f2c767adcbee272637ab048e16f26c840f7abc)